File: Leetcode-one-by-one/src/Determine Whether Matrix Can Be Obtained By Rotation.go
func findRotation(mat [][]int, target [][]int) bool {
    for i := 0; i < 4; i++ {
        if isEqual(mat, target) {
            return true
        }
        mat = rotate(mat)
    }
    return false
}

func isEqual(a [][]int, b [][]int) bool {
    n := len(a)
    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            if a[i][j] != b[i][j] {
                return false
            }
        }
    }
    return true
}

func rotate(mat [][]int) [][]int {
    n := len(mat)
    res := make([][]int, n)
    for i := range res {
        res[i] = make([]int, n)
    }

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            res[j][n-1-i] = mat[i][j]
        }
    }
    return res
}

File: Leetcode-one-by-one/src/Determine Whether Matrix Can Be Obtained By Rotation.swift
class Solution {
    func findRotation(_ mat: [[Int]], _ target: [[Int]]) -> Bool {
        var mat = mat
        
        for _ in 0..<4 {
            if isEqual(mat, target) {
                return true
            }
            mat = rotate(mat)
        }
        return false
    }
    
    private func isEqual(_ a: [[Int]], _ b: [[Int]]) -> Bool {
        let n = a.count
        for i in 0..<n {
            for j in 0..<n {
                if a[i][j] != b[i][j] {
                    return false
                }
            }
        }
        return true
    }
    
    private func rotate(_ mat: [[Int]]) -> [[Int]] {
        let n = mat.count
        var res = Array(repeating: Array(repeating: 0, count: n), count: n)
        
        for i in 0..<n {
            for j in 0..<n {
                res[j][n - 1 - i] = mat[i][j]
            }
        }
        return res
    }
}
